functions.php

```diff
@@ -53,6 +53,10 @@ function as_schedule_single_action( $timestamp, $hook, $args = array(), $group =
  * @return int The action ID.
  */
 function as_schedule_recurring_action( $timestamp, $interval_in_seconds, $hook, $args = array(), $group = '', $unique = false ) {
+	// if you pass a string by mistake, it causes a flood of scheduled actions, recurring almost every minute or less
+	if ( ! is_int( $interval_in_seconds ) ) {
+		throw new Exception( "Interval parameter should be of type integer, received" . gettype( $interval_in_seconds ) );
```

Comment:
Similar to the earlier comment, we may wish to take a softer stance. If there is existing code out there that is inadvertently making a mistake, we don't necessarily want to end the request with an exception. Possible iteration:

```suggestion
		_doing_it_wrong(
			__METHOD__,
			sprintf( 
				/* translators: 1: provided value 2: provided type. */
				__( 'An integer was expected but "%1$s" (%2%s) was received.', 'action-scheduler' ),
				$interval_in_seconds,
				gettype( $interval_in_seconds )
			),
			ActionScheduler_Versions::instance()->latest_version()
		);
```

This would also make the exception message translatable.